solutions/ipetek/scripts/BagToLaserScan.cpp

```diff
@@ -0,0 +1,44 @@
+#define _USE_MATH_DEFINES
+#include <ros/ros.h>
+#include <sensor_msgs/LaserScan.h>
+#include <visualization_msgs/Marker.h>
+
+#include <cmath>
+#include <vector>
+
+class BagToLaserScan {
+   public:
+    BagToLaserScan(const ros::NodeHandle& nh) : nh_(nh) {
+        bagSub_ = nh_.subscribe("/points", 1, &BagToLaserScan::bagCallback, this);
+        laserPub_ = nh_.advertise<sensor_msgs::LaserScan>("/laser", 1);
+
+        pub_msg_.angle_min = 0;
+        pub_msg_.angle_max = 2 * M_PI;
+        pub_msg_.range_max = 100;
+        pub_msg_.angle_increment = 2 * M_PI / (10000 - 1);
+        pub_msg_.header.frame_id = "image";
+    }
+
+   private:
+    ros::NodeHandle nh_;
+    ros::Subscriber bagSub_;
+    ros::Publisher laserPub_;
+    sensor_msgs::LaserScan pub_msg_;
+
+    void bagCallback(const visualization_msgs::Marker::ConstPtr msg) {
+        pub_msg_.ranges = std::vector<_Float32>(0);
```

Comment:
If we already have a vector with a certain capacity, it is preferable to call `.clear()`. This will avoid redundant memory allocations.

(If there isn't a previous vector available, good practice is to `.reserve()` the expected size before inserting the elements. This also helps avoid redundant memory reallocations.)

Reply:
Marking as resolved :)